docs/how_to/deploy/adreno.rst

```diff
@@ -0,0 +1,259 @@
+-  TP has a dedicated L1 cache, which is read-only cache and stores data
+   fetched from level-2 (L2) cache for texture operations (primary
+   reason)
+
+-  Handling of image boundaries is built-in.
+
+-  Supports numerous image format and data type combinations with
+   support for automatic format conversions
+
+Multiple texturing or multitexturing is the use of more than one texture
+at a time on a polygon. Adreno GPUs support up to 32 total textures in a
+single render pass, i.e., up to 16 textures in the fragment shader and
+up to 16 textures at a time for the vertex shader.
+
+Effective use of multiple textures significantly reduces overdraw, saves
+ALU cost for fragment shaders, and avoids unnecessary vertex transforms.
+
+With textures, it is possible to achieve a significant performance boost
+compared to OpenCL buffer based solutions.
+
+Building TVM for Adreno
+-----------------------
+
+This section gives instructions on how to build the Android part of TVM
+via OpenCL-SDK and TVM RPC Server in order to deploy models on Adreno.
+
+Since the process of building TVM for Adreno is exactly the same as the
+process of building TVM for Android, please refer to these instructions:
+`TVM RPC
+Server <https://github.com/apache/tvm/tree/main/apps/cpp_rpc>`__
+Alternatively, to build a TVM via docker using OpenCL-Headers and set-up
+with Android TVM RPC, refer to this guide: `Deploy the Pretrained Model
+on
+Android <https://tvm.apache.org/docs/how_to/deploy_models/deploy_model_on_android.html>`__
+
+**Prerequisites**: Android NDK, Android Debug Bridge (adb), OpenCL
+
+For us to begin with, Android NDK, Android Debug Bridge and OpenCL must
+be installed and Android part of TVM must be builded.
+
+**Android NDK installation**: https://developer.android.com/ndk
+
+**Android Debug Bridge installation**:
+https://developer.android.com/studio/command-line/adb
+
+**OpenCL installation**: https://github.com/KhronosGroup/OpenCL-SDK.git
+
+You can also build the android part of TVM via console. From the root
+folder of TVM:
+
+::
+
+   mkdir build_android
+   cd build_android
+   cmake .. -DUSE_OPENCL=path/to/OpenCL -DCMAKE_TOOLCHAIN_FILE=${ANDROID_NDK_HOME}/build/cmake/android.toolchain.cmake -DANDROID_ABI=arm64-v8a -DANDROID_NATIVE_API_LEVEL=android-28 -DCMAKE_FIND_ROOT_PATH_MODE_PACKAGE=ON -DANDROID_STL=c++_static -DUSE_CPP_RPC=ON
+   make -jN tvm_runtime tvm_rpc
+
+where **N** is the number of cores available on your *CPU*.
+
+At this stage you have builded TVM for Adreno.
+
+Build and deploy model for Adreno
+---------------------------------
+
+As the deployment pipelines on Android and on Adreno do not have many
+differences, in this section we will focus on one and only necessary
+aspect to compile and deploy models for Adreno and will take a look at
+the generation of kernels with and without textures. In addition, the
+possibility of choosing a different precision for model compilation will
+be considered.
+
+| |Android deployment pipeline|
```

Comment:
Align all down arrows to have the same sizes here:
<img width="336" alt="image" src="https://user-images.githubusercontent.com/5525113/197483405-71363f04-37f1-428a-bf0f-df0f7a851130.png">

Be sure that all arrows are direct and don't have angles like here: 
<img width="640" alt="image" src="https://user-images.githubusercontent.com/5525113/197483616-b6bd309b-00a4-4210-ad62-59f90e755348.png">